freshmaker/handlers/koji/rebuild_flatpak_application_on_module_ready.py

```diff
@@ -20,13 +20,150 @@
+            msg = "There is no image can be rebuilt. " f"message_id: {event.msg_id}"
+            db_event.transition(EventState.SKIPPED, msg)
+            db.session.commit()
+            self.log_info(msg)
+            return []
+
+        images_nvr = ",".join([rebuild_image.nvr for rebuild_image in rebuild_images])
+        self.log_info("Following images %s will be rebuilt" % images_nvr)
+
+        # TODO: Return empty list so far as this method is still in progress.
+        # Will think about whether to return the real BaseEvent objects in future.
+        return []
+
+    def _get_requests_session(cls, retry_options={}):
+        """
+        Create a requests session. Reference from:
+        https://github.com/release-engineering/cachito/blob/master/cachito/workers/requests.py
+
+        :param dict retry_options: overwrite options for initialization of Retry instance
+        :return: the configured requests session
+        :rtype: requests.Session
+        """
+        DEFAULT_RETRY_OPTIONS = {
+            "total": 5,
+            "read": 5,
+            "connect": 5,
+            "backoff_factor": 1.3,
+            "status_forcelist": (500, 502, 503, 504),
+        }
+        session = requests.Session()
+        retry_options = {**DEFAULT_RETRY_OPTIONS, **retry_options}
+        adapter = HTTPAdapter(max_retries=Retry(**retry_options))
+        session.mount("http://", adapter)
+        session.mount("https://", adapter)
+        return session
+
+    def _get_auto_rebuild_image_list(self, event):
+        """
+        Get rebuild image list which can be auto rebuilt.
+
+        :param event FlatpakModuleAdvisoryReadyEvent: The event this handler
+            is currently handling.
+        :rtype: list
+        :return: List of images which can be auto rebuilt.
+        """
+        if not conf.pyxis_server_url:
+            raise ValueError("'PYXIS_SERVER_URL' parameter should be set")
+        self._pyxis = Pyxis(conf.pyxis_server_url)
+
+        if not conf.flatpak_server_url:
+            raise ValueError(
+                "'FLATPAK_SERVER_URL' parameter should be set"
+            )
+
+        errata = Errata()
+        module_nvrs = errata.get_cve_affected_build_nvrs(event.advisory.errata_id, True)
+        rebuild_images_list = list()
+
+        with koji_service(
+            conf.koji_profile, log, login=False, dry_run=self.dry_run
+        ) as session:
+            for module_nvr in module_nvrs:
+                mmd = session.get_modulemd(module_nvr)
+                content_index_url = "{}/released/contents/modules/{}:{}.json".format(
+                    conf.flatpak_server_url,
+                    mmd.get_module_name(),
+                    mmd.get_stream_name(),
+                )
+                req_session = self._get_requests_session()
+                response = req_session.get(content_index_url)
+                status_code = response.status_code
+                if status_code == HTTPStatus.OK:
```

Comment:
Probably we can just check `response.ok` (`True` if `status_code` is less than 400) instead of comparing the status code with `HTTPStatus.OK` (value == 200).
```suggestion
                if response.ok:
```

Reply:
Ack. Fixing